lib/srv/sess.go

```diff
@@ -820,6 +821,38 @@ func (s *session) startInteractive(ch ssh.Channel, ctx *ServerContext) error {
 			ctx.Errorf("Failed to close enhanced recording (interactive) session: %v: %v.", s.id, err)
 		}
 
+		if ctx.ExecRequest.GetCommand() != "" {
+			execEvent := &events.Exec{
```

Comment:
Can you move this to a separate function to keep the scope of the parent function small?